bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpModeForwardFilterTest.java

```diff
@@ -15,181 +15,112 @@
-          .thenReturn(this.pageMock);
-        when(this.pageMock.getProperties())
-          .thenReturn(this.mapSample);
-
-
-        when(this.slingHttpServletRequestMock.getRequestDispatcher(any(Resource.class), any(RequestDispatcherOptions.class)))
-          .thenReturn(null);
-
-        this.amff.doFilter(this.servletRequestMock, this.servletResponseMock, this.filterChainMock);
-
-        assertThat(this.testLogger.getLoggingEvents(), hasItem(debug("Request dispatcher is null. AMP mode forwarding aborted.")));
-        verify(this.filterChainMock, times(1)).doFilter(this.servletRequestMock, this.servletResponseMock);
+    void testFilteringWithNoSelector() throws IOException, ServletException {
+        context.currentPage(AMP_PAGE_PROPERTY);
+        context.currentResource();
+        spyAmpModeForwardFilter.doFilter(context.request(), context.response(), filterChain);
+        verify(spyAmpModeForwardFilter, atLeastOnce()).doFilter(eq(context.request()), eq(context.response()) , eq(filterChain));
     }
 
     @Test
-    public void doFilter_ampOnlyForwardFalse() throws IOException, ServletException {
-        Map<String, Object> map = new HashMap<String, Object>(){{
-            put(AMP_MODE_PROP, "ampOnly");
-        }};
-        this.mapSample = new ValueMapDecorator(map);
-
-        when(this.slingHttpServletRequestMock.getRequestPathInfo())
-          .thenReturn(this.requestPathInfoMock);
-        when(this.requestPathInfoMock.getSelectorString())
-          .thenReturn("");
-
-        when(this.slingHttpServletRequestMock.getResourceResolver())
-          .thenReturn(this.resourceResolverMock);
-        when(this.slingHttpServletRequestMock.getResource())
-          .thenReturn(this.resourceMock);
-        when(this.resourceResolverMock.adaptTo(PageManager.class))
-          .thenReturn(this.pageManagerMock);
-        when(this.pageManagerMock.getContainingPage(this.resourceMock))
-          .thenReturn(this.pageMock);
-        when(this.pageMock.getProperties())
-          .thenReturn(this.mapSample);
-
-
-        when(this.slingHttpServletRequestMock.getRequestDispatcher(any(Resource.class), any(RequestDispatcherOptions.class)))
-          .thenReturn(null);
-
-        this.amff.doFilter(this.servletRequestMock, this.servletResponseMock, this.filterChainMock);
-
-        assertThat(this.testLogger.getLoggingEvents(), hasItem(debug("Request dispatcher is null. AMP mode forwarding aborted.")));
-        verify(this.filterChainMock, times(1)).doFilter(this.servletRequestMock, this.servletResponseMock);
+    void testFilteringWithAmpSelector() throws IOException, ServletException {
+        //without amp selector
+        context.currentPage(AMP_SELECTOR);
+        spyAmpModeForwardFilter.doFilter(context.request(), context.response(), filterChain);
+        verify(spyAmpModeForwardFilter, atLeastOnce()).doFilter(eq(context.request()), eq(context.response()) , eq(filterChain));
```

Comment:
these `verify()` methods are good, but you should also be doing assertions after these because we need to make sure the processing itself is happening correctly, not just that it's happening. that's what the ArgumentCaptor should allow you to do

Reply:
Thanks for reviewing it. The comment makes sense, and I tried doing some assertion checks, but there is not a request or response transformation logic in the Filter that I can look for to do assertion checks. The filter checks the amp mode, adds the `RequestDispactherOptions`, and then dispatches the request using the `dispatcher.forward()`.